src/middlewares/auth.ts

```diff
@@ -0,0 +1,63 @@
+import { NextFunction, Request, Response } from 'express';
+import { ForbiddenError, UnauthorizedError } from './error-handler.js';
+import jwt from 'jsonwebtoken';
+import memoize from 'memoizee';
+import axios from 'axios';
+import ms from 'ms';
+import * as config from '../config.js';
+import {egoUrl} from "../config.js";
+
+const getKey = memoize(
+	async (egoURL: string) => {
+		const response = await axios.get(egoURL);
+		return response.data;
+	},
+	{
+		maxAge: ms('1h'),
```

Comment:
Let's add a config property for `AUTH_PUBLICKEY_CACHE` with a default as 1 day. This let's people override this without code change if ever needed.

Also ignore this if there is another PR pending for auth changes.

Reply:
will take this up in the auth PR